src/main/java/org/elasticsearch/transport/netty/NettyTransport.java

```diff
@@ -1117,6 +1120,45 @@ private void closeChannelsAndWait(Channel[] channels, List<ChannelFuture> future
                     //ignore
                 }
             }
+            for (ChannelFuture future : futures) {
+                future.awaitUninterruptibly();
+            }
+        }
+    }
+
+    class ScheduledPing implements Runnable {
+
+        volatile int successfulPings;
+        volatile int failedPings;
+
+        @Override
+        public void run() {
+            if (lifecycle.stoppedOrClosed()) {
+                return;
+            }
+            for (Map.Entry<DiscoveryNode, NodeChannels> entry : connectedNodes.entrySet()) {
```

Comment:
can you assign the key and the value here before we use it? it's way easier to read